src/sentry/static/sentry/app/components/avatarSettings.jsx

```diff
@@ -0,0 +1,515 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+import AlertActions from '../actions/alertActions';
+import ApiMixin from '../mixins/apiMixin';
+import LoadingError from '../components/loadingError';
+import LoadingIndicator from '../components/loadingIndicator';
+import {t} from '../locale';
+
+
+const AvatarCropper = React.createClass({
+  propTypes: {
+    user: React.PropTypes.object.isRequired,
+    updateDataUrlState: React.PropTypes.func.isRequired,
+    savedDataUrl: React.PropTypes.string
+  },
+
+  mixins: [ApiMixin],
+
+  getInitialState() {
+    return {
+      error: null,
+      mousePosition: {
+        pageX: null,
+        pageY: null
+      },
+      resizeDimensions: {
+        top: 0,
+        left: 0,
+        size: 0
+      }
+    };
+  },
+
+  MIN_DIMENSION: 256,
+
+  MAX_DIMENSION: 1024,
+
+  onChange(ev) {
+    let file = ev.target.files[0];
+    if (!/^image\//.test(file.type)) {
+      this.setState({error: 'That is not a supported file type.'});
+      return;
```

Comment:
Not sure if you're aware, but a little trick you can do to remove a separate line for `return` is:

```javascript
return void this.setState({error: 'That is not a supported file type.'});
```

This causes the return value from this.setState to be ignored and `undefined` is returned instead (just like `return` on an empty line).

The nice thing about it is that you no longer need braces in your `if` statement since it's now a single statement (if you care – I think we mostly wrap with braces anyways).